core/corehttp/gateway_handler.go

```diff
@@ -875,3 +743,110 @@ func fixupSuperfluousNamespace(w http.ResponseWriter, urlPath string, urlQuery s
 		ErrorMsg:      fmt.Sprintf("invalid path: %q should be %q", urlPath, intendedPath.String()),
 	}) == nil
 }
+
+func validateSupportedHeaders(w http.ResponseWriter, r *http.Request) bool {
+	// X-Ipfs-Gateway-Prefix was removed (https://github.com/ipfs/go-ipfs/issues/7702)
+	// TODO: remove this after  go-ipfs 0.13 ships
+	if prfx := r.Header.Get("X-Ipfs-Gateway-Prefix"); prfx != "" {
+		err := fmt.Errorf("X-Ipfs-Gateway-Prefix support was removed: https://github.com/ipfs/go-ipfs/issues/7702")
+		webError(w, "unsupported HTTP header", err, http.StatusBadRequest)
+		return false
```

Comment:
As noted above, I'm not so sure I'm writing idiomatic Go here.  At the very least it would probably be good to have a more consistent naming convention for the functions such that it was clear whether true or false meant there was an error and we should return.